### PR TITLE
fix cancellation directions

### PR DESCRIPTION
**Pull request by paasovaara**
- added integration tests 
- fix for cancellation message directions https://github.com/HSLdevcom/transitdata-tripupdate-processor/issues/42
- fix for train filtering https://github.com/HSLdevcom/transitdata-tripupdate-processor/issues/43

### CODE REVIEW

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java [hunk 1]
```diff
@@ -74,10 +74,10 @@ private GtfsRtFactory() {
     }
 
     public static GtfsRealtime.TripUpdate newTripUpdate(InternalMessages.TripCancellation cancellation, long timestampMs) {
-
+        final int gtfsRtDirection = StopEvent.pubtransDirectionToGtfsDirection(cancellation.getDirectionId());
```
Comment (kenkus-futurice) [inline src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: Could be renamed to `StopEvent.joreDirectionToGtfsDirection`?

File: src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java [hunk 2]
```diff
@@ -149,7 +180,24 @@ public void testImpl(TestPipeline.TestContext context) throws Exception {
...
+                GtfsRealtime.TripUpdate tu = feedMessage.getEntity(0).getTripUpdate();
+                assertNotNull(tu);
+
+                assertEquals(1, tu.getStopTimeUpdateCount());
+                GtfsRealtime.TripUpdate.StopTimeUpdate update = tu.getStopTimeUpdate(0);
+                assertNotNull(update);
+                assertTrue(update.hasArrival());
+
+                // We add departure also which should be the same as arrival?
+                //assertFalse(update.hasDeparture());
```
Comment (kenkus-futurice) [inline src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java]: This assert is not needed?

File: src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java (hunk 1 above)
Comment (paasovaara) [reply src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java]: good point 👍

File: src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java (hunk 2 above)
Comment (paasovaara) [reply src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java]: actually this should have it's own test completely. Thanx for noticing.

Comment (paasovaara) [reply src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java]: Validated from code that yes, we always have both arrival and departure. -> added testcase for both arrival and departure events. 

requires some refactoring but I'll do that in my next pr which is on the way.

**Pull request merged**